types/ramda/test/mapObjIndexed-tests.ts

```diff
@@ -11,14 +11,47 @@ import * as R from 'ramda';
 };
 
 () => {
-  const testObject: {
-    [key: string]: Error;
-  } = {
-    hello: new Error('hello'),
-  };
-  const errorMessages = R.mapObjIndexed(function test(value, key) {
-    // value should be inferred.
-    return value.message + String(key);
-  }, testObject);
-  console.log(errorMessages);
+    const testObject: {
+        [k: string]: Error;
+    } = {
+        hello: new Error('hello'),
+    };
+    const errorMessages = R.mapObjIndexed(function test(value, key) {
+        // value should be inferred.
+        const test = key.toLowerCase();
+        return value.message + String(key);
+    }, testObject);
+    console.log(errorMessages);
+};
+
+() => {
+    type TestKeys = 'hello' | 'test';
+    const testObject: Record<TestKeys, string> = {
+        hello: 'world',
+        test: '123',
+    };
+
+    const mapped = R.mapObjIndexed(function test(value, key) {
+        function unreachableCase(v: never): never {
+            throw new Error('not possible');
+        }
+
+        let returnValue: number;
+        // type of key should be inferred as `TestKeys` (not `string`).
+        if (key === 'hello') {
+            returnValue = 42;
+        } else if (key === 'test') {
+            returnValue = 13;
+        } else {
+            // make sure the type system knows that the key can't be anything other than 'hello' | 'test'
+            return unreachableCase(key);
+        }
+
+        const test1 = value.charAt(0);
+        return returnValue;
```

Comment:
This would probably be much easier to test if you used `// $ExpectType` directives, but I appreciate that you’ve demonstrated the type without that here.